pkg/api/validate/vm.go

```diff
@@ -98,7 +98,7 @@ func DiskSizeIsValid(sizeGB int) bool {
 
 func VMSizeIsValid(vmSize api.VMSize, requiredD2sV3Workers, isMaster bool) bool {
```

Comment:
I'm curious, why did you choose to expose `SupportedMasterVmSizes` instead of just using this function?

Reply:
I'm going back and forth on this. `VMSizeIsValid` is case sensitive and checks some other things we don't care about here, so it might be better to just look directly at the list rather than make VMSizeIsValid work in this context. But I also think it would be nice to just use the single function for both, we'd have to tweak it though. Thoughts @petrkotas @s-amann?

Reply:
This seems to work without breaking unit tests, which is equivalent to what you have in the frontend
```
func VMSizeIsValid(vmSize api.VMSize, requiredD2sV3Workers, isMaster bool) bool {
	if isMaster {
		for k := range SupportedMasterVmSizes {
			if strings.EqualFold(string(k), string(vmSize)) {
				return true
			}
		}
	}
```

I do agree there is a tradeoff here, as this function is used in each API Version.